src/pages/workspaces/workspace/analysis/modals/AnalysisDuplicator.test.ts

```diff
@@ -0,0 +1,307 @@
+  lastModified: new Date().getTime(),
+  cloudProvider
+})
+
+const baseTestFile: AnalysisFile = getTestFile('test/file0.ipynb' as AbsolutePath)
+
+const onDismiss = jest.fn()
+const onSuccess = jest.fn()
+
+const defaultModalProps: AnalysisDuplicatorProps = {
+  destroyOld: false,
+  fromLauncher: false,
+  printName: baseTestFile.fileName,
+  toolLabel: toolLabelTypes[tools.Jupyter.label],
+  workspaceInfo: defaultGoogleWorkspace.workspace,
+  onDismiss, onSuccess
+}
+
+describe('AnalysisDuplicator', () => {
+  beforeEach(() => {
+    jest.resetAllMocks()
+    // Arrange
+    const googleStorageMock: Partial<GoogleStorageContract> = ({
+      listAnalyses: jest.fn(() => Promise.resolve([]))
+    })
+    asMockedFn(GoogleStorage).mockImplementation(() => googleStorageMock as GoogleStorageContract)
+
+    const azureStorageMock: Partial<AzureStorageContract> = ({
+      listNotebooks: jest.fn()
+    })
+    asMockedFn(AzureStorage).mockImplementation(() => azureStorageMock as AzureStorageContract)
+
+    asMockedFn(useAnalysisFiles).mockImplementation(() => ({
+      refresh: () => Promise.resolve(),
+      loadedState: { state: [], status: 'Ready' }
+    }))
+  })
+
+  it('renders correctly by default with destroyOld false', () => {
+    // Arrange
+
+    // Act
+    render(h(AnalysisDuplicator, defaultModalProps))
+
+    // Assert
+    screen.getByText(`Copy "${baseTestFile.fileName}"`)
+  })
+
+  it('renders correctly by default with destroyOld true', () => {
+    // Arrange
+
+    // Act
+    render(h(AnalysisDuplicator, {
+      ...defaultModalProps,
+      destroyOld: true
+    }))
+
+    // Assert
+    screen.getByText(`Rename "${baseTestFile.fileName}"`)
+  })
+
+  it.each(
+    [
+      { inputText: ' ', errorMsg: 'Name can\'t be blank' },
+      { inputText: 'invalid$', errorMsg: 'Name can\'t contain these characters:' },
+      { inputText: 'file1', errorMsg: 'Name already exists' }
+    ]
+  )('rejects existing and invalid names', async ({ inputText, errorMsg }) => {
+    // Arrange
+    const fileList = [getTestFile('test/file1.ipynb' as AbsolutePath), getTestFile('test/file2.ipynb' as AbsolutePath)]
+    asMockedFn(useAnalysisFiles).mockImplementation(() => ({
+      loadedState: { state: fileList, status: 'Ready' },
+      refresh: () => Promise.resolve(),
+    }))
+
+    // Act
+    act(() => {
+      render(h(AnalysisDuplicator, defaultModalProps))
+    })
+
+    // Assert
+    const input = screen.getByLabelText(/New Name/)
+    await userEvent.type(input, inputText)
+
+    expect(screen.getAllByText(errorMsg).length).toBe(2)
+    const button = screen.getByText('Copy Analysis')
+    expect(button).toHaveAttribute('disabled')
+  })
+
+  it('copies for a google workspace correctly', async () => {
+    // Arrange
+    const fileList = [getTestFile('test/file1.ipynb' as AbsolutePath), getTestFile('test/file2.ipynb' as AbsolutePath)]
+    const calledMock = jest.fn()
```

Comment:
Suggest naming mocks after the function they're mocking. `expect(copy).toHaveBeenCalled` is clearer than `expect(calledMock).toHaveBeenCalled`.